src/fireedge/src/modules/components/Tabs/Quota/Components/QuotaControls.js

```diff
@@ -0,0 +1,470 @@
+import { memo, useMemo, useEffect, useState, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Autocomplete,
+  TextField,
+  Grid,
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  Box,
+  Typography,
+  useTheme,
+} from '@mui/material'
+
+import {
+  UserAPI,
+  GroupAPI,
+  useGeneralApi,
+  SystemAPI,
+  ClusterAPI,
+  DatastoreAPI,
+  ImageAPI,
+  VnAPI,
+} from '@FeaturesModule'
+
+import { T, STYLE_BUTTONS } from '@ConstantsModule'
+import { Tr } from '@modules/components/HOC'
+import {
+  validateResourceId,
+  validateValue,
+  useQuotaControlReducer,
+  getConcatenatedValues,
+  quotaIdentifiers,
+  handleApplyGlobalQuotas,
+} from '@modules/components/Tabs/Quota/Components/helpers/scripts'
+
+import {
+  HybridInputField,
+  ResourceIDAutocomplete,
+} from '@modules/components/Tabs/Quota/Components/helpers/subcomponents'
+
+import { SubmitButton } from '@modules/components/FormControl'
+
+import { mapValues, map } from 'lodash'
+
+/**
+ * QuotaControls Component
+ *
+ * @param {object} props - Props for the component
+ * @param {Array} props.quotaTypes - Available quota types
+ * @param {string} props.userId - User ID
+ * @param {string} props.selectedType - Selected quota type
+ * @param {Function} props.setSelectedType - Function to set selected quota type
+ * @param {Array} props.existingResourceIDs - Existing resource IDs
+ * @param {object} props.clickedElement - Clicked element data.
+ * @param {object} props.nameMaps - Resource name mappings.
+ */
+export const QuotaControls = memo(
+  ({
+    quotaTypes,
+    userId,
+    selectedType,
+    setSelectedType,
+    existingData,
+    clickedElement,
+    nameMaps,
+    groups,
+  }) => {
+    const [getClusters] = ClusterAPI.useLazyGetClustersQuery()
+    const [getDatastores] = DatastoreAPI.useLazyGetDatastoresQuery()
+    const [getImages] = ImageAPI.useLazyGetImagesQuery()
+    const [getNetworks] = VnAPI.useLazyGetVNetworksQuery()
+
+    const [state, actions] = useQuotaControlReducer()
+    const [popoverAnchorEl, setPopoverAnchorEl] = useState(null)
+    const [existingResourceIds, setExistingResourceIds] = useState(null)
+    const [touchedFields, setTouchedFields] = useState({})
+    const { enqueueError, enqueueSuccess } = useGeneralApi()
+    const { data: { QUOTA_VM_ATTRIBUTE: genericQuotas = [] } = {} } =
+      SystemAPI.useGetOneConfigQuery()
+
+    const formatGenericQuotas = (
+      Array.isArray(genericQuotas) ? genericQuotas : [genericQuotas]
+    )?.reduce((acc, quota) => {
+      acc.push(
+        {
+          id: quota,
+          displayName: quota.charAt(0) + quota.slice(1).toLowerCase(),
+        },
+        {
+          id: `RUNNING_${quota?.toUpperCase()}`,
+          displayName: `Running ${
+            quota.charAt(0) + quota.slice(1).toLowerCase()
+          }`,
+        }
+      )
+
+      return acc
+    }, [])
+
+    const extendedQuotaIdentifiers = mapValues(quotaIdentifiers, (quotaArray) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

CSRF protection missing for state-changing requests. Implement CSRF tokens for POST/PUT/PATCH/DELETE operations.